pkg/rpc/context.go

```diff
@@ -1013,7 +1015,8 @@ func (a internalClientAdapter) MuxRangeFeed(
 		sender:   eventWriter,
 	}
 
-	go func() {
+	wg := ctxgroup.WithContext(serverCtx)
+	wg.Go(func() error {
```

Comment:
You are probably right. I just want to learn more here. Why don't we need to use `wg.GoCtx` to watch for this context shutdown in https://github.com/cockroachdb/cockroach/blob/beeec160a11848d750c2cc77237781f0137d181e/pkg/util/ctxgroup/ctxgroup.go#L179? Will wg goroutines (server side call) shut down on itself when the wg context is canceled?

Reply:
Note that nothing inside this goroutine takes a context.  I haven't looked at the full implementation, but that is the main reason I didn't see a point in using GoCtx.  I assume that shutdown happens on cancellation by virtue of the fact that we store `serverCtx` in rawServerStream and that is handling it correctly. But I didn't fully verify this.

Reply:
`sp.IsNoop()` check  `ss.Context()`, which would originally return client passed ctx but now returns the server context https://github.com/cockroachdb/cockroach/blob/f20164ebb1d5021f2294b8683f1d7dd21e283881/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L149. 

Can the interceptors end up using the already canceled `serverCtx` during the server side shutdown?https://github.com/cockroachdb/cockroach/blob/f20164ebb1d5021f2294b8683f1d7dd21e283881/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L158-L160 

Or are these interceptors entirely bypassed when serverCtx is canceled and the server stream shut down?

Reply:
Discussed this offline - the tracing span is actually from the client side so it is fine to cancel server context early here.